tools/src/main/java/org/apache/kafka/trogdor/coordinator/TaskManager.java

```diff
@@ -316,42 +377,13 @@ public void createTask(final String id, TaskSpec spec)
         }
 
         @Override
-        public Void call() throws Exception {
-            if (id.isEmpty()) {
-                throw new InvalidRequestException("Invalid empty ID in createTask request.");
-            }
-            ManagedTask task = tasks.get(id);
-            if (task != null) {
-                if (!task.spec.equals(spec)) {
-                    throw new RequestConflictException("Task ID " + id + " already " +
-                        "exists, and has a different spec " + task.spec);
-                }
-                log.info("Task {} already exists with spec {}", id, spec);
-                return null;
-            }
-            TaskController controller = null;
-            String failure = null;
+        public ManagedTask call() throws Exception {
             try {
-                controller = spec.newController(id);
-            } catch (Throwable t) {
-                failure = "Failed to create TaskController: " + t.getMessage();
+                TaskController controller = spec.newController(id);
+                return new ManagedTask(id, spec, controller, TaskStateType.PENDING);
+            } catch (Exception t) {
+                throw new Exception("Failed to create TaskController: " + t.getMessage());
```

Comment:
This shouldn't throw an exception.  The existing logic where we fail the task is what we want to do here.